kotsadm/pkg/handlers/identity.go

```diff
@@ -0,0 +1,224 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"github.com/dexidp/dex/connector/oidc"
+	"github.com/gosimple/slug"
+	"github.com/replicatedhq/kots/kotsadm/pkg/logger"
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+	"github.com/replicatedhq/kots/pkg/identity"
+	"github.com/replicatedhq/kots/pkg/ingress"
+	"github.com/replicatedhq/kots/pkg/kotsadm"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+type ConfigureIdentityServiceRequest struct {
+	AdminConsoleAddress    string `json:"adminConsoleAddress"`
+	IdentityServiceAddress string `json:"identityServiceAddress"`
+	IdentityProvider       string `json:"identityProvider"`
+	Issuer                 string `json:"issuer"`
+	ClientID               string `json:"clientId"`
+	ClientSecret           string `json:"clientSecret"`
+}
+
+type ConfigureIdentityServiceResponse struct {
+	Success bool   `json:"success"`
+	Error   string `json:"error,omitempty"`
+}
+
+func ConfigureIdentityService(w http.ResponseWriter, r *http.Request) {
+	response := ConfigureIdentityServiceResponse{}
+
+	request := ConfigureIdentityServiceRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
+		logger.Error(err)
+		response.Error = "failed to decode request body"
+		JSON(w, http.StatusInternalServerError, response)
+		return
+	}
+
+	namespace := os.Getenv("POD_NAMESPACE")
+
+	ingressConfig, err := ingress.GetConfig(r.Context(), namespace)
+	if err != nil {
+		logger.Error(err)
+		response.Error = "failed to get ingress config"
+		JSON(w, http.StatusInternalServerError, response)
+		return
+	}
+
+	// TODO: support types other than oidc
+	connectorConfig := oidc.Config{
+		Issuer:                    request.Issuer,
+		ClientID:                  request.ClientID,
+		ClientSecret:              request.ClientSecret,
+		RedirectURI:               "{{OIDCIdentityCallbackURL}}",
+		GetUserInfo:               true,
```

Comment:
this may not be available from their oidc provider. this should be an advanced option with default true